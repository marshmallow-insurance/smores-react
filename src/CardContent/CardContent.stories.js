import React from 'react';
import { storiesOf } from '@storybook/react';

import { text } from '@storybook/addon-knobs';

import Card from '../Card';
import Text from '../Text';
import CardContent from './CardContent';

storiesOf('CardContent', module)
  .addWithJSX('default', () => (
    <Card>
      <CardContent>
        <Text tag='h1' weight='bold'>Default padding</Text>
      </CardContent>
    </Card>
  ))
  .addWithJSX('narrow', () => (
    <Card>
      <CardContent narrow>
        <Text tag='h3' weight='bold'>Narrow padding</Text>
      </CardContent>
    </Card>
  ))