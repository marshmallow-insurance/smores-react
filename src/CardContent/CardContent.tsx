import React, {FC, ReactNode} from 'react';
import styled from 'styled-components';

interface ICardContent {
  narrow: boolean;
}

type Props = {
  children: ReactNode;
  className?: string;
  /** variation with narrow padding */
  narrow?: boolean;
};

export const CardContent: FC<Props> = ({
  children,
  className,
  narrow = false,
}) => (
  <Container className={className} narrow={narrow}>
    {children}
  </Container>
);

const Container = styled.div<ICardContent>`
  padding: ${p => (p.narrow ? '8px' : '32px')};

  @media (min-width: 768px) {
    padding: ${p => (p.narrow ? '16px' : '48px')};
  }
`;
