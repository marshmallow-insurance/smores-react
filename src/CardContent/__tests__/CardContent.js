import React from 'react';
import {render} from 'react-testing-library';
import 'jest-styled-components';

import {CardContent} from '../CardContent';

test('renders', () => {
  const {container} = render(<CardContent>Text inside</CardContent>);
  expect(container.firstChild).toMatchSnapshot();
});
