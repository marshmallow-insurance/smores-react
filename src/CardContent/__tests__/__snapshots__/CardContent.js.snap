// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
.c0 {
  padding: 32px;
}

@media (min-width:768px) {
  .c0 {
    padding: 48px;
  }
}

<div
  class="c0"
>
  Text inside
</div>
`;
